fedocal/__init__.py

```diff
@@ -864,9 +870,30 @@ def add_meeting(calendar_name):
             day=form.meeting_date.data.day))
     elif flask.request.method == 'GET':
         form = forms.AddMeetingForm(timezone=tzone, calendars=calendars)
+        mtg_date = flask.request.args.get('date', None)
+        if mtg_date:
+            form.meeting_date.data = parser.parse(mtg_date).date()
+            form.meeting_date_end.data = parser.parse(mtg_date).date()
+
+        mtg_time = flask.request.args.get('time', None)
+        if mtg_time:
+            form.meeting_time_start.data = mtg_time
+            hours, minutes = mtg_time.split(':')[:2]
+            try:
+                hours = int(hours) + 1
```

Comment:
So the user click on the agenda matrix, the time is retrieved by JS, given to the URL called, we parse it here to do a small data checking, then it is put in the form.
In the add_meeting template we extract the value from the form (if there is one) and insert it into the timespinner().

This way, if you click on a meeting at 18:00, the default time you get are 18:00 start - 19:00 end, but this does not prevent you in anyway from adjusting the times as you would currently.